### ISSUE TITLE
brick_server new client API - rename

### ISSUE THREAD

**Issue by norton** [hibari/gdss-brick#2]
Rename a OldKey/Value pair to Key/Value pair in a brick, failing if OldKey does not already exist or if Key already exists.  Similar to get_many, the rename API only works for keys of the same chain.

**Comment by tatsuya6502** [comment]
I have decided to include this feature (`rename/6`) into upcoming Hibari v0.1.11 release. I modified the codes on the dev branch so that brick_ets will act like hlog sequences are always frozen. I'll revisit this in next unstable major release v0.3.

[Commit](https://github.com/hibari/gdss-brick/commit/0483490a669c659794fab57ae51c0506c09f9d12)
> gdss-brick >> GH2 - brick_server new client API - rename
>
> rename/6 operation now always uses `rename_key_with_get_set_delete/6` instead of `my_insert/6` with `?KEY_SWITCHAPOO`. Will revisit this in Hibari v0.3 release.

**Comment by tatsuya6502** [comment]
Changing the milestone from v0.3 to v0.1.11.

**Comment by tatsuya6502** [comment]
> Commit
>
> gdss-brick >> GH2 - brick_server new client API - rename
>
> rename/6 operation now always uses rename_key_with_get_set_delete/6 instead of my_insert/6 with ?KEY_SWITCHAPOO. Will revisit this in Hibari v0.3 release.

QuickCheck property `prop_simple1` has found a bug in this implementation. If new key is overwriting an existing key-value with `{exp_time_directive, keep}` and `{attrib_directive, keep}`, `rename_key_with_get_set_delete/6` will not only keep the exp_time and flags from old key (correct), but also from the existing key-value (wrong).

```
[{set,{var,1},
      {call,simple_eqc_tests,simple_set,
            [["/foo/bar/",["000",50]],
             <<"v a">>,1426323338786789025,
             [{exp_time_directive,keep}],
             [{attrib2,"val2"}]]}},
 {set,{var,2},
      {call,simple_eqc_tests,simple_set,
            [<<"/foo/bar/0001">>,<<"v a">>,1426323338789594039,
             [{exp_time_directive,replace},{attrib_directive,keep}],
             []]}},
 {set,{var,3},
      {call,simple_eqc_tests,simple_rename,
            [<<"/foo/bar/0001">>,
             ["/foo/bar/",["000",50]],
             1426323338789723040,
             [{exp_time_directive,keep},{attrib_directive,keep}],
             []]}},
 {set,{var,4},
      {call,simple_eqc_tests,simple_get,["/foo/bar/0002",[get_all_attribs]]}}]
S = {state,5,
           [{<<"/foo/bar/0001">>,undefined},
            {<<"/foo/bar/0002">>,1426323338789594039},
            {<<"/foo/bar/0003">>,undefined},
            {<<"/foo/bar/0004">>,undefined}],
           [{<<"/foo/bar/0001">>,[]},
            {<<"/foo/bar/0002">>,[<<"v a">>]},
            {<<"/foo/bar/0003">>,[]},
            {<<"/foo/bar/0004">>,[]}],
           [{<<"/foo/bar/0001">>,undefined},
            {<<"/foo/bar/0002">>,[]},
            {<<"/foo/bar/0003">>,undefined},
            {<<"/foo/bar/0004">>,undefined}],
           [{tab1_ch1_b1,gdss_eunit@localhost}]}
R = {postcondition,false}
Hist = [{eqc_statem_history,
            {state,1,
                [{<<"/foo/bar/0001">>,undefined},
                 {<<"/foo/bar/0002">>,undefined},
                 {<<"/foo/bar/0003">>,undefined},
                 {<<"/foo/bar/0004">>,undefined}],
                [{<<"/foo/bar/0001">>,[]},
                 {<<"/foo/bar/0002">>,[]},
                 {<<"/foo/bar/0003">>,[]},
                 {<<"/foo/bar/0004">>,[]}],
                [{<<"/foo/bar/0001">>,undefined},
                 {<<"/foo/bar/0002">>,undefined},
                 {<<"/foo/bar/0003">>,undefined},
                 {<<"/foo/bar/0004">>,undefined}],
                [{tab1_ch1_b1,gdss_eunit@localhost}]},
            {call,simple_eqc_tests,simple_set,
                [["/foo/bar/",["000",50]],
                 <<"v a">>,1426323338786789025,
                 [{exp_time_directive,keep}],
                 [{attrib2,"val2"}]]},
            [],
            {normal,{ok,1426323340565845}}},
        {eqc_statem_history,
            {state,2,
                [{<<"/foo/bar/0001">>,undefined},
                 {<<"/foo/bar/0002">>,1426323338786789025},
                 {<<"/foo/bar/0003">>,undefined},
                 {<<"/foo/bar/0004">>,undefined}],
                [{<<"/foo/bar/0001">>,[]},
                 {<<"/foo/bar/0002">>,[<<"v a">>]},
                 {<<"/foo/bar/0003">>,[]},
                 {<<"/foo/bar/0004">>,[]}],
                [{<<"/foo/bar/0001">>,undefined},
                 {<<"/foo/bar/0002">>,[{attrib2,"val2"}]},
                 {<<"/foo/bar/0003">>,undefined},
                 {<<"/foo/bar/0004">>,undefined}],
                [{tab1_ch1_b1,gdss_eunit@localhost}]},
            {call,simple_eqc_tests,simple_set,
                [<<"/foo/bar/0001">>,<<"v a">>,1426323338789594039,
                 [{exp_time_directive,replace},{attrib_directive,keep}],
                 []]},
            [],
            {normal,{ok,1426323340572921}}},
        {eqc_statem_history,
            {state,3,
                [{<<"/foo/bar/0001">>,1426323338789594039},
                 {<<"/foo/bar/0002">>,1426323338786789025},
                 {<<"/foo/bar/0003">>,undefined},
                 {<<"/foo/bar/0004">>,undefined}],
                [{<<"/foo/bar/0001">>,[<<"v a">>]},
                 {<<"/foo/bar/0002">>,[<<"v a">>]},
                 {<<"/foo/bar/0003">>,[]},
                 {<<"/foo/bar/0004">>,[]}],
                [{<<"/foo/bar/0001">>,[]},
                 {<<"/foo/bar/0002">>,[{attrib2,"val2"}]},
                 {<<"/foo/bar/0003">>,undefined},
                 {<<"/foo/bar/0004">>,undefined}],
                [{tab1_ch1_b1,gdss_eunit@localhost}]},
            {call,simple_eqc_tests,simple_rename,
                [<<"/foo/bar/0001">>,
                 ["/foo/bar/",["000",50]],
                 1426323338789723040,
                 [{exp_time_directive,keep},{attrib_directive,keep}],
                 []]},
            [],
            {normal,{ok,1426323340580630}}},
        {eqc_statem_history,
            {state,4,
                [{<<"/foo/bar/0001">>,undefined},
                 {<<"/foo/bar/0002">>,1426323338789594039},
                 {<<"/foo/bar/0003">>,undefined},
                 {<<"/foo/bar/0004">>,undefined}],
                [{<<"/foo/bar/0001">>,[]},
                 {<<"/foo/bar/0002">>,[<<"v a">>]},
                 {<<"/foo/bar/0003">>,[]},
                 {<<"/foo/bar/0004">>,[]}],
                [{<<"/foo/bar/0001">>,undefined},
                 {<<"/foo/bar/0002">>,[]},
                 {<<"/foo/bar/0003">>,undefined},
                 {<<"/foo/bar/0004">>,undefined}],
                [{tab1_ch1_b1,gdss_eunit@localhost}]},
            {call,simple_eqc_tests,simple_get,
                ["/foo/bar/0002",[get_all_attribs]]},
            [],
            {normal,
                {ok,1426323340580630,<<"v a">>,1426323338786789025,
                    [{val_len,3},{attrib2,"val2"}]}}}]
```

**Comment by tatsuya6502** [comment]
> QuickCheck property `prop_simple1` has found a bug in this implementation. If new key is overwriting an existing key-value with {exp_time_directive, keep} and {attrib_directive, keep}, rename_key_with_get_set_delete/6 will not only keep the exp_time and flags from old key (correct), but also from the existing key-value (wrong).

Fixed this by https://github.com/hibari/gdss-brick/commit/8c7fb13097f0cba0db49f5754a26849747ed59cb.
Now `prop_simple1` passes 2,000 tests.

```erl
Erlang/OTP 17 [erts-6.3] [source] [64-bit] [smp:8:8] [async-threads:64] [hipe] [kernel-poll:true]

Eshell V6.3  (abort with ^G)
(hibariclient@127.0.0.1)1> eqc:quickcheck(eqc:numtests(2000, simple_eqc_tests:prop_simple1())).


Hibari: get_many() has been commented out from command()

Starting Quviq QuickCheck version 1.33.3
   (compiled at {{2015,2,9},{14,18,24}})
...
vY.YY.Yv....c.....c....Y.......cv.Y.v..c.Y....v....Y.v......c.c..
...
OK, passed 2000 tests

49.45% {0,div10}
25.00% {1,div10}
12.20% {2,div10}
5.75% {3,div10}
3.05% {4,div10}
2.00% {5,div10}
0.85% {6,div10}
0.80% {7,div10}
0.40% {9,div10}
0.20% {10,div10}
0.20% {8,div10}
0.05% {16,div10}
0.05% {11,div10}
true
(hibariclient@127.0.0.1)2> 
```

**Comment by tatsuya6502** [comment]
> Fixed this by 8c7fb13.
> Now `prop_simple1` passes 2,000 tests.

Applied the same fix to branch `gbrick-gh17-redesign-disk-storage`: https://github.com/hibari/gdss-brick/commit/e9957c5527d3b7e570e554947b0e74e69560276c
